test/instances/eeth/EETHHyperdrive.t.sol

```diff
@@ -0,0 +1,618 @@
+            EETHConversions.convertToBase(
+                POOL,
+                IERC20(address(EETH)),
+                shareAmount
+            );
+    }
+
+    /// @dev Deploys the Morpho Blue deployer coordinator contract.
+    /// @param _factory The address of the Hyperdrive factory.
+    function deployCoordinator(
+        address _factory
+    ) internal override returns (address) {
+        vm.startPrank(alice);
+        return
+            address(
+                new EETHHyperdriveDeployerCoordinator(
+                    string.concat(__testConfig.name, "DeployerCoordinator"),
+                    _factory,
+                    address(new EETHHyperdriveCoreDeployer(POOL)),
+                    address(new EETHTarget0Deployer(POOL)),
+                    address(new EETHTarget1Deployer(POOL)),
+                    address(new EETHTarget2Deployer(POOL)),
+                    address(new EETHTarget3Deployer(POOL)),
+                    address(new EETHTarget4Deployer(POOL)),
+                    POOL
+                )
+            );
+    }
+
+    /// @dev Fetches the total supply of the base and share tokens.
+    function getSupply() internal view override returns (uint256, uint256) {
+        return (POOL.getTotalPooledEther(), EETH.totalShares());
+    }
+
+    /// @dev Fetches the token balance information of an account.
+    function getTokenBalances(
+        address account
+    ) internal view override returns (uint256, uint256) {
+        return (EETH.balanceOf(account), EETH.shares(account));
+    }
+
+    /// @dev Verifies that deposit accounting is correct when opening positions.
+    function verifyDeposit(
+        address trader,
+        uint256 amountPaid,
+        bool asBase,
+        uint256 totalBaseBefore,
+        uint256 totalSharesBefore,
+        AccountBalances memory traderBalancesBefore,
+        AccountBalances memory hyperdriveBalancesBefore
+    ) internal view override {
+        if (asBase) {
+            // Ensure that the amount of pooled ether increased by the base paid.
+            assertEq(POOL.getTotalPooledEther(), totalBaseBefore + amountPaid);
+
+            // Ensure that the ETH balances were updated correctly.
+            assertEq(
+                address(hyperdrive).balance,
+                hyperdriveBalancesBefore.ETHBalance
+            );
+            assertEq(bob.balance, traderBalancesBefore.ETHBalance - amountPaid);
+
+            // Ensure that the EETH balances were updated correctly.
+            assertApproxEqAbs(
+                EETH.balanceOf(address(hyperdrive)),
+                hyperdriveBalancesBefore.baseBalance + amountPaid,
+                1
+            );
+            assertEq(EETH.balanceOf(trader), traderBalancesBefore.baseBalance);
+
+            // Ensure that the EETH shares were updated correctly.
+            uint256 expectedShares = amountPaid.mulDivDown(
+                totalSharesBefore,
+                totalBaseBefore
+            );
```

Comment:
We should be able to use the conversion for this when we update it to use `mulDivDown`